epub33/core/index.html

```diff
@@ -1244,12 +1244,22 @@ <h5>Shared Attributes</h5>
 								<code>dir</code>
 							</dt>
 							<dd>
-								<p>Specifies the base text direction of the content and attribute values of the carrying
-									element and its descendants.</p>
-								<p>Inherent directionality specified using [[!Unicode]] takes precedence over this
-									attribute.</p>
-								<p>Allowed values are <code>ltr</code> (left-to-right) and <code>rtl</code>
-									(right-to-left).</p>
+								<p>Specifies the base direction [[!BIDI]] of the carrying element and its
+									descendants.</p>
+								<p>Allowed values are:</p>
+								<ul>
+									<li><code>ltr</code> &#8212; left-to-right base direction;</li>
+									<li><code>rtl</code> &#8212; right-to-left base direction; and</li>
+									<li><code>auto</code> &#8212; base direction is determined using the Unicode Bidi
+										Algorithm [[BIDI]].</li>
+								</ul>
+								<p>The value <code>auto</code> is assumed when the attribute is not specified or
+									contains an invalid value.</p>
+								<div class="note">
+									<p>The base direction does not override the inherent direction of text runs within
+										an element but specifies how those runs are ordered relative to each other for
+										display.</p>
```

Comment:
I'm not sure what "inherent direction" means here. Do you mean the [logical order](https://www.w3.org/TR/charmod/#sec-LogicalOrder) of the characters?

Reply:
Thanks, yes, that's looks like the better technical phrasing for it.

Reply:
I'm not sure that's correct either – depending on what you really wanted to say.

The **logical** order of Unicode text describes the order of characters in storage, which is unidirectional, and ordered essentially as the user would speak or type the text.  There is no bidi text or RTL text in storage. Everything is unidirectional.

However, to render the text you can imagine that it goes through two steps: 
- first, sequences of characters with the same _strong_ Unicode direction property are arranged into 'runs' in which the characters progress according to that direction, eg. a word in Hebrew containing only Hebrew characters has characters that proceed from right to left.
- second, the ordering of those directional runs relative to each other and the positioning of characters with only weak directional properties are determined by the base direction that is set.

For examples, see [Unicode Bidirectional Algorithm basics](https://www.w3.org/International/articles/inline-bidi-markup/uba-basics.en).

So to say that the dir attribute or base direction doesn't affect the logical order of text kind of goes without saying, because that's a fundamental Unicode rule.  But the intention might have been to say that the `dir` attribute or base direction doesn't affect the ordering of characters within directional runs, but does affect the relative ordering of those runs and the placement weak directional characters such as punctuation.

Does that help?

Reply:
> So to say that the dir attribute or base direction doesn't affect the logical order of text kind of goes without saying, because that's a fundamental Unicode rule.

I don't get the feeling any of us are completely sure what this statement was trying to get at:

> Inherent directionality specified using [Unicode] takes precedence over this attribute.

I think it may have been saying what you write in the former statement, but what you've expressed in the latter makes more sense. Any objections to adapting that as the note text, so modifying it to:

> The base direction specified in the `dir` attribute does not affect the ordering of characters within directional runs, only the relative ordering of those runs and the placement weak directional characters such as punctuation.

Reply:
Sounds good to me.

Reply:
Are you okay with the new wording @xfq ?

Reply:
Looks good to me.